src/NetTopologySuite/Geometries/GeometryFactory.cs

```diff
@@ -68,6 +68,26 @@ public class GeometryFactory
         /// </summary>
         public int SRID => _srid;
 
+        private GeometryOverlay _geometryOverlay;
+
+        /// <summary>
+        /// Gets a value indicating the geometry overlay function set to use
+        /// </summary>
+        internal GeometryOverlay GeometryOverlay
+        {
+            get { return _geometryOverlay ?? (_geometryOverlay = NtsGeometryServices.Instance.GeometryOverlay); }
+            set
+            {
+                if (_geometryOverlay != null)
+                    throw new InvalidOperationException("GeometryOverlay already set.");
+
+                if (value == null)
+                    value = NtsGeometryServices.Instance.GeometryOverlay;
+
+                _geometryOverlay = value;
+            }
```

Comment:
Now that this is internal, I think we can simplify without losing anything important:
```csharp
get { return _geometryOverlay ?? NtsGeometryServices.Instance.GeometryOverlay; }
set { _geometryOverlay = value; }
```